packages/flutter/lib/src/material/material_state.dart

```diff
@@ -301,9 +301,19 @@ typedef MaterialStateTextStyle = WidgetStateTextStyle;
 /// [MaterialStateOutlineInputBorder] and override its [resolve] method. You'll also need
 /// to provide a `defaultValue` to the super constructor, so that we can know
 /// at compile-time what its default color is.
+@Deprecated(
+  'Use WidgetStateInputBorder instead. '
+  'Renamed to match other WidgetStateProperty objects. '
+  'This feature was deprecated after v3.26.0-0.1.pre.'
+)
 abstract class MaterialStateOutlineInputBorder extends OutlineInputBorder implements MaterialStateProperty<InputBorder> {
```

Comment:
Also, to proactively answer a question before it's asked…

I considered matching the other types in this file, something like:

```dart
typedef MaterialStateOutlineInputBorder = WidgetStateInputBorder;
typedef MaterialStateUnderlineInputBorder = WidgetStateInputBorder;
```

<br>

The only problem with that is when someone makes a custom subclass:

```dart
class MyBorder extends MaterialStateOutlineInputBorder {} // works

class MyBorder extends WidgetStateInputBorder {} // doesn't work

class MyBorder extends OutlineInputBorder implements WidgetStateInputBorder {
  // makes no difference whether OutlineInputBorder or UnderlineInputBorder is extended
}
```

So instead we have a data-driven fix for the `.resolveWith()` constructor, and documentation that explains how to make a subclass. (Perhaps I could also update the [MaterialState migration guide](https://github.com/flutter/website/blob/main/src/content/release/breaking-changes/material-state.md) to include this info.)